bundles/org.openhab.binding.speedtest/src/main/java/org/openhab/binding/speedtest/internal/SpeedtestHandler.java

```diff
@@ -324,58 +322,67 @@ private void getSpeed() {
+        newState = new QuantityType<>(Double.parseDouble(downloadElapsed) / 1000.0, Units.SECOND);
+        logger.debug("downloadElapsed: {}", newState);
         updateState(new ChannelUID(getThing().getUID(), SpeedtestBindingConstants.DOWNLOAD_ELAPSED), newState);
 
-        newState = new QuantityType<>(Double.parseDouble(upload_bandwidth) / 125000.0, Units.MEGABIT_PER_SECOND);
-        logger.debug("upload_bandwidth: {}", newState);
+        newState = new QuantityType<>(Double.parseDouble(uploadBandwidth) / 125000.0, Units.MEGABIT_PER_SECOND);
+        logger.debug("uploadBandwidth: {}", newState);
         updateState(new ChannelUID(getThing().getUID(), SpeedtestBindingConstants.UPLOAD_BANDWIDTH), newState);
 
-        newState = new QuantityType<>(Double.parseDouble(upload_bytes), Units.BYTE);
-        logger.debug("upload_bytes: {}", newState);
+        newState = new QuantityType<>(Double.parseDouble(uploadBytes), Units.BYTE);
+        logger.debug("uploadBytes: {}", newState);
         updateState(new ChannelUID(getThing().getUID(), SpeedtestBindingConstants.UPLOAD_BYTES), newState);
 
-        newState = new QuantityType<>(Double.parseDouble(upload_elapsed) / 1000.0, Units.SECOND);
-        logger.debug("upload_elapsed: {}", newState);
+        newState = new QuantityType<>(Double.parseDouble(uploadElapsed) / 1000.0, Units.SECOND);
+        logger.debug("uploadElapsed: {}", newState);
         updateState(new ChannelUID(getThing().getUID(), SpeedtestBindingConstants.UPLOAD_ELAPSED), newState);
 
         updateState(new ChannelUID(getThing().getUID(), SpeedtestBindingConstants.INTERFACE_EXTERNALIP),
-                new StringType(String.valueOf(interface_externalIp)));
+                new StringType(String.valueOf(interfaceExternalIp)));
         updateState(new ChannelUID(getThing().getUID(), SpeedtestBindingConstants.INTERFACE_INTERNALIP),
-                new StringType(String.valueOf(interface_internalIp)));
+                new StringType(String.valueOf(interfaceInternalIp)));
         updateState(new ChannelUID(getThing().getUID(), SpeedtestBindingConstants.ISP),
                 new StringType(String.valueOf(isp)));
         updateState(new ChannelUID(getThing().getUID(), SpeedtestBindingConstants.RESULT_URL),
-                new StringType(String.valueOf(result_url)));
+                new StringType(String.valueOf(resultUrl)));
         updateState(new ChannelUID(getThing().getUID(), SpeedtestBindingConstants.SERVER),
-                new StringType(String.valueOf(server_name)));
+                new StringType(String.valueOf(server)));
     }
 
     /*
      * Checks to make sure the executable for speedtest is valid
      */
     public boolean checkConfig(String execPath) {
-        if (checkExecPath(execPath)) { // Check if entered path exists
-            return true;
+        File file = new File(execPath);
+        if (checkFileExists(file)) { // Check if entered path exists
```

Comment:
thingstatus should be set when the check fails. I think you forgot the !

Reply:
Sorry, you are right. I forgot the "!" and as it seems the testing of the latest change 👎